backend/src/modules/notifications/listeners/others.listener.ts

```diff
@@ -1,21 +1,52 @@
 import { Injectable } from '@nestjs/common';
 import { OnEvent } from '@nestjs/event-emitter';
+import { VolunteerStatus } from 'src/modules/volunteer/enums/volunteer-status.enum';
+import { VolunteerFacade } from 'src/modules/volunteer/services/volunteer.facade';
+import { GetOneAnnouncementUseCase } from 'src/usecases/announcement/get-one-announcement.usecase';
+import { GetManyVolunteersUseCase } from 'src/usecases/volunteer/get-many-volunteers.usecase';
 import { EVENTS } from '../constants/events.constants';
 import SendAnnouncementEvent from '../events/others/send-announcement.event';
 
 @Injectable()
 export class OthersListener {
+  constructor(
+    private readonly getOneAnnouncementUseCase: GetOneAnnouncementUseCase,
+    private readonly getManyVolunteersUseCase: GetManyVolunteersUseCase,
+    private readonly volunteerFacade: VolunteerFacade,
+  ) {}
+
   @OnEvent(EVENTS.OTHER.SEND_ANNOUNCEMENT)
   async sendSendAnnouncementEvent(
     payload: SendAnnouncementEvent,
   ): Promise<void> {
-    console.log(payload);
-    // TODO: 1. Retrieve the organization data
-    // TODO: 2. Throw exception if organization not found
-    // TODO: 3. Retrive the announcement data
-    // TODO: 4. Throw exception if announcement not found
-    // TODO: 5. Retrieve the target data
-    // TODO: 6. Throw exception if target not found
-    // TODO: 7. Send push notification to target with announcement link
+    const { organizationId, announcementId, targetIds } = payload;
+    // 1. Retrive the announcement data
+    const announcement = await this.getOneAnnouncementUseCase.execute({
```

Comment:
What happens if one of these queries fails?

Reply:
Also i am not sure about using usecases here as they are throwing HttpException whereas here is not the case.